### PR TITLE
Highlight max combo on beatmap leaderboards

### LINKED ISSUE #7862: Max combo's aren't green highlighted when someone has a full combo on it
**Screenshots or videos showing encountered issue:** 

lazer:
osu!_DfZGXP1u0U
web:
chrome_fKY8i83t8i

**osu!lazer version:** 
commit e80c310ab29030fd4d6c5d300a98f8f51c68067c

**Used beatmap:** https://osu.ppy.sh/beatmapsets/999868#osu/2091238

### PR DESCRIPTION
**Pull request by thewildtree**
Fixes #7862 

Made `MaxCombo` nullable because there were situations where `max_combo` returned from the server was null and caused the beatmap overlay to load infinitely.

dotnet_2020-02-19_17-53-26

### CODE REVIEW

File: osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs [hunk 1]
```diff
@@ -146,7 +146,8 @@ private Drawable[] createContent(int index, ScoreInfo score)
                 new OsuSpriteText
                 {
                     Text = $@"{score.MaxCombo:N0}x",
-                    Font = OsuFont.GetFont(size: text_size)
+                    Font = OsuFont.GetFont(size: text_size),
+                    Colour = score.MaxCombo == score.Beatmap.MaxCombo ? highAccuracyColour : Color4.White
```
Comment (bdach) [inline osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs]: Better null handling is required (as evidenced by tests failing).

**Review by bdach — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs (hunk 1 above)
Comment (TheWildTree) [reply osu.Game/Overlays/BeatmapSet/Scores/ScoreTable.cs]: woops, fixed

**Review by bdach — APPROVED** [review]
At a closer look `TestSceneScoreContainer` should probably provide the previously-null beatmap, but since API docs are incomplete I can't even really tell if the beatmap is mandatory or not in the actual web responses without digging in PHP sources (which is risky at best for me).

Maybe better to stay on the safe side for now and potentially clean up when `ScoreContainer` is fixed up.

**Review by smoogipoo — APPROVED** [review]
Hmm... I'm a little suspicious of the null still, but I've confirmed with the osu-web team and it can be null if it's awaiting difficulty processing, etc.

**Pull request merged**